apps/system/js/remote/system_remote.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+(function(exports) {
+  var SystemRemote = {
+    DEBUG: false,
+
+    REQUIRED_MODULES: ['SimpleAppWindowManager', 'WallpaperManager'],
+    REGISTERED_EVENTS: ['wallpaperchange'],
+
+    _started: false,
+    _readyCallback: null,
+
+    debug: function(msg) {
+      if (this.DEBUG && msg) {
+        this.printLog.apply(this, arguments);
+      }
+    },
+
+    printLog: function() {
+      var arg = ['[SystemRemote #' + this.displayId + '] ' + arguments[0]]
+        .concat(Array.from(arguments).slice(1));
+      console.log.apply(console, arg);
+    },
+
+    ready: function(callback) {
+      if (this._started) {
+        setTimeout(callback);
+      } else {
+        if (!this._readyCallback) {
+          this._readyCallback = [];
+        }
+        this._readyCallback.push(callback);
+      }
+    },
+
+    start: function() {
+      if (this._started) {
+        return;
+      }
+
+      this._started = true;
+
+      if (window.location.hash) {
+        this.displayId = parseInt(window.location.hash.substring(1), 10);
+      }
+
+      this.debug('Starting');
+
+      this.REQUIRED_MODULES.forEach((module) => {
```

Comment:
Not really. I just want to avoid repeated `var xxx = new XXX; xxx.start();` patterns here but BaseModule/Service seems like overkill to our current need.